### PR TITLE
Autofix Rubocop Gemspec/DeprecatedAttributeAssignment

### PR DESCRIPTION
**Pull request by nschonni**
https://docs.rubocop.org/rubocop/cops_gemspec.html#gemspecdeprecatedattributeassignment

### CODE REVIEW

**Review by ohler55 — APPROVED** [review]

**Comment by ohler55** [comment]
Instead of many tiny PRs can you batch them up?

**Comment by nschonni** [comment]
Sure, do you want me to just keep batching up rules in this one and then you can decide if you want to squash them or rebase out some?

**Comment by ohler55** [comment]
That would be great. If it gets too large then we can merge and get on this the next one. Thanks for all this work.

**Comment by nschonni** [comment]
I ignored the TODO, and ran `rubocop -x` to autofix all but some of the LineLength rules at once. Maybe you don't like some of the changes, and I can try and figure out which of https://docs.rubocop.org/rubocop/cops_layout.htm to just turn off instead of autofix

File: ext/ox/extconf.rb [hunk 1]
```diff
@@ -19,15 +19,15 @@
   'RUBY_VERSION_MICRO' => version[2],
 }
 
-dflags.each do |k,v|
+dflags.each do |k, v|
```
Comment (ohler55) [inline ext/ox/extconf.rb:22]: This is an odd format rule. Can the rule be made to keep k and v on the same line like other code changes?

Comment (nschonni) [reply ext/ox/extconf.rb:22]: Do you mean, no space after a comma? I'm not seeing a multi-line change here

File: test/sax/sax_test.rb [hunk 2]
```diff
@@ -395,17 +387,15 @@ def test_sax_open_close_element
...
   end
 
   def test_sax_empty_element
     Ox::default_options = $ox_sax_options
     parse_compare(%{  <top>  </top>},
                   [[:start_element, :top],
                    [:text, " "],
-                   [:end_element, :top]
-                  ], AllSax, { :skip => :skip_white })
+                   [:end_element, :top]], AllSax, { :skip => :skip_white })
```
Comment (ohler55) [inline test/sax/sax_test.rb:398]: I prefer the original. It's easier to read and easier to edit if new items are to be added from the list.

File: ext/ox/extconf.rb (hunk 1 above)
Comment (ohler55) [reply ext/ox/extconf.rb:22]: That's odd. In the GitHub viewer the was on a separate line. It's fine the way it appears here.

**Comment by ohler55** [comment]
One thing I like to do is make function argument initial values be tighter like `foo={}` instead of `foo = {}`. Not a big deal if it isn't easy to configure and I'll gladly bend but of possible...

There seems to be some rules that kick in on the some of the lists/arrays in the tests. In comment on one but there are many more. The rule that does that is not something I would want to enforce but instead deprecate.

**Comment by nschonni** [comment]
OK, I'll go back and just go rule-by-rule so it's easier to tell what is changing what, and if there is an override that's preferred

**Comment by nschonni** [comment]
Would you prefer if all the rules in the `test` folder get ignored for now? I could exclude it, then regenerate the todo

**Comment by ohler55** [comment]
That might make it easier to see the changes.

BTW, Layout/SpaceAroundEqualsInParameterDefault is the rule for argument default I think.

**Comment by nschonni** [comment]
I rebased out the big cleaning, and submitted it rule by rule for the ones that are more on/off to see if they look OK before looking at the customizable ones

**Comment by ohler55** [comment]
LGTM

Merge this and start another?

**Comment by nschonni** [comment]
Sure, I can do the same thing with a handful of rules on the next one, then it's easier to disable or set preperences on particular ones

**Comment by ohler55** [comment]
The PR seems like about the right size.

Day is almost done for me. At least no snow is expected tonight. :-) (Huntsville ON)

**Comment by nschonni** [comment]
Ottawa only go a little dusted :)
I'll do one more tonight, but don't worry about reviewing till whenever

**Comment by ohler55** [comment]
ok, will look it over in the morning if I can resist looking tonight.

**Pull request merged**